init.c

```diff
@@ -73,6 +73,8 @@ bool write_bw_log = false;
 bool read_only = false;
 int status_interval = 0;
 
+int ss_check_interval = 1000;
```

Comment:
The new `thread_options` member is unsigned but this is signed. Wouldn't it better to make this unsigned as well for the sake of consistency?

Also, there are multiple places where this is divided by or multiplied by 1000. Consider making this `ss_check_interval_us` so that the units are clear and we can avoid some arithmetic.